src/lib/modes.js

```diff
@@ -10,6 +10,7 @@ export const IDENT_RE = '[a-zA-Z]\\w*';
 export const UNDERSCORE_IDENT_RE = '[a-zA-Z_]\\w*';
 export const NUMBER_RE = '\\b\\d+(\\.\\d+)?';
 export const C_NUMBER_RE = '(-?)(\\b0[xX][a-fA-F0-9]+|(\\b\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)'; // 0x..., 0..., decimal, float
+export const EXTENDED_NUMBER_RE = '([-+]?)(\\b0[xX][a-fA-F0-9]+|(\\b\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)|NaN|[-+]?Infinity'; // 0x..., 0..., decimal, float
```

Comment:
Is this all valid ECMAScript?  If so perhaps this belongs in `lib/ecmascript`, but I don't think we want to add another global here since this is a very specific use case.